lib/register.php

```diff
@@ -269,6 +269,10 @@ function gutenberg_can_edit_post( $post ) {
 		return false;
 	}
 
+	if ( $post->ID == get_option( 'page_for_posts' ) && empty( $post->post_content ) ) {
```

Comment:
- Needs [Yoda condition](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) (see failing build)
- Should ideally be using strict equality. This isn't codified in the standard, but if we expect it to take a particular value shape, we'd be better off than to allow leniency (prone to false positives)
- The condition is not self-explanatory, and could do for an inline code comment explaining why it exists.